include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,116 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#define RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#include <type_traits>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        namespace view
+        {
+            /// \addtogroup group-views
+            /// @{
+
+            template <typename T>
+            struct linear_distribute_view
+              : view_facade<linear_distribute_view<T>, finite>
+            {
+                CONCEPT_ASSERT(std::is_arithmetic<T>());
+            private:
+                friend range_access;
+
+                T from_, to_;
+                std::ptrdiff_t n_;
+                
+                constexpr T read() const noexcept
+                {
+                    return from_;
+                }
+                constexpr bool equal(default_sentinel) const noexcept
+                {
+                    return n_ == 0;
+                }
+                constexpr bool equal(linear_distribute_view const& other) const noexcept
+                {
```

Comment:
`equal(view const&)` is called by `operator==(basic_iterator<view> const&, basic_iterator<view> const&)`. You can support (slowly) comparing iterators from different ranges by returning `std::tie(from_, to_, n_) == std::tie(that.from_, that.to_, that.n_)`, or you can declare that only iterators from the same range are in the domain of equality and return `(RANGES_EXPECT(std::tie(from_, to_) == std::tie(that.from_, that.to_)), n_ == that.n_)` or `(RANGES_EXPECT(std::tie(to_, n_) == std::tie(that.to_, that.n_)), from_ == that.from_)`. The flavor that compares only a single integer will probably be the fastest in release builds. In the name of "don't pay for what you don't use," range-v3 always chooses speed for comparison with ranges over enlarging the domain of `==`. (It would be awesome if STL2 were to have distinct hooks for "fast comparison" using operators, and "total ordering" using some other customization point so that we could usefully provide both operations on iterators.)

`distance_to(view const& that)` is called by `operator-(basic_iterator<view> const&, basic_iterator<view> const&)`. Similar reasoning applies here, although I think you would have a much harder time making this operation work across distinct ranges than `equal`.